### PR TITLE
overwrite v2 visitors

### PR DESCRIPTION
**Pull request by pawel-big-lebowski**
Signed-off-by: Pawel Leszczynski <leszczynski.pawel@gmail.com>

### Problem

Spark visitors for `OverwriteByExpression` and `OverwritePartitionsDynamic`

Closes: #ISSUE-NUMBER

### Solution

Implement visitors, make implementation consistent to use StorageHandlers to implement locations' logic

> **Note:** All schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [ ] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

If you're contributing a new integration, please specify the scope of the integration and how/where it has been tested (e.g., Apache Spark integration supports `S3` and `GCS` filesystem operations, tested with AWS EMR).

### Checklist

- [X] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [X] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [X] Your changes are accompanied by tests (_if relevant_)
- [X] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] You've updated the [`CHANGELOG.md`](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md) with details about your change under the "Unreleased" section (_if relevant, depending on the change, this may not be necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)

### CODE REVIEW

File: integration/spark/src/test/common/java/io/openlineage/spark/agent/SparkContainerIntegrationTest.java [hunk 1]
```diff
@@ -78,7 +78,7 @@ public void cleanupSpark() {
       logger.error("Unable to shut down pyspark container", e2);
     }
     try {
-      kafka.stop();
+      if (kafka != null) kafka.stop();
```
Comment (mobuchowski) [inline integration/spark/src/test/common/java/io/openlineage/spark/agent/SparkContainerIntegrationTest.java:81]: 👍

File: integration/spark/src/main/spark3/java/io/openlineage/spark3/agent/lifecycle/plan/catalog/IcebergHandler.java [hunk 2]
```diff
@@ -45,7 +47,7 @@ public DatasetIdentifier getDatasetIdentifier(
     String prefix = String.format("spark.sql.catalog.%s", catalogName);
     Map<String, String> conf =
         ScalaConversionUtils.<String, String>fromMap(session.conf().getAll());
-    log.warn(conf.toString());
+    log.info(conf.toString());
```
Comment (mobuchowski) [inline integration/spark/src/main/spark3/java/io/openlineage/spark3/agent/lifecycle/plan/catalog/IcebergHandler.java:50]: 👍

**Review by mobuchowski — CHANGES_REQUESTED** [review]

File: integration/spark/src/main/spark3/java/io/openlineage/spark3/agent/utils/PlanUtils3.java [hunk 3]
```diff
@@ -36,6 +41,39 @@
...
+
+    return Collections.singletonList(PlanUtils.getDataset(datasetIdentifier, schema, facets));
+  }
+
+  public static List<OpenLineage.Dataset> fromDataSourceV2Relation(
+      SparkSession session, DataSourceV2Relation relation) {
+    return fromDataSourceV2Relation(session, relation, new HashMap<>());
+  }
+
+  public static List<OpenLineage.Dataset> fromDataSourceV2Relation(
```
[the remark this answers is not in the document]
Comment (pawel-big-lebowski) [reply integration/spark/src/main/spark3/java/io/openlineage/spark3/agent/utils/PlanUtils3.java:58]: Indeed, let's merge the other one first and try to rebase this one onto PR447.

[the remark this answers is not in the document]
Comment (pawel-big-lebowski) [reply integration/spark/src/main/spark3/java/io/openlineage/spark3/agent/utils/PlanUtils3.java]: I rebased the branch onto `spark_extension_impl` to solve the conflicts.

**Pull request closed without merging**